Lib/pathlib.py

```diff
@@ -1361,13 +1361,23 @@ def lstat(self):
             self._raise_closed()
         return self._accessor.lstat(self)
 
-    def link_to(self, target):
+    def link_to(self, link):
         """
-        Create a hard link pointing to a path named target.
+        Make *link* a hard link pointing to the same file as this path.
+
+        Deprecated since Python 3.9, use `hardlink_to()`.
+        """
+        if self._closed:
+            self._raise_closed()
+        self._accessor.link(self, link)
+
+    def hardlink_to(self, target):
+        """
+        Make this path a hard link pointing to the same file as the given path.
```

Comment:
Probably worth adding a note like in `symlink_to()`'s docstring that the order is reversed as compared to `os.link()`.